Improve db configuration checks

GernotMaier opened the pull request:
This is a minor update of inconsistent behavior when running simtools with incomplete database configuration. Essentially the code didn't notice at the configuration step that something was wrong, but much later when trying to call the DB.

Two changes:

- added a jsonschema based validation of the DB configuration dictionary in db_handler. This handles also the two optional parameter correctly
- fix a small bug in the setting of the default db_configuration in the configurator: if any parameter was missing (even an optional one), an error was thrown.

EshitaJoshi commented on src/simtools/db/db_handler.py:
```diff
@@ -24,20 +25,47 @@
 
 
+jsonschema_db_dict = {
+    "$schema": "https://json-schema.org/draft-07/schema#",
```
I seem to get a warning from the `jsonschema` library `DeprecationWarning: The metaschema specified by $schema was not found. Using the latest draft to validate, but this will raise an error in the future.` Do you get this too?

GernotMaier replied:
Thanks a lot for that - I didn't see the warning when I worked on the code (I can only see it when I run pytest with the "-s" option.

I've updated the jsonschema version throughout the code (in some schema files unrelated to the DB).

Fixed also one issue noticed now when running tests without a local '.env' file (which we anyway shouldn't; but let's keep the code robust)

orelgueta commented on src/simtools/db/db_handler.py:
```diff
@@ -24,20 +25,47 @@
+    "description": "MongoDB configuration",
+    "properties": {
+        "db_server": {"type": "string", "description": "DB server address"},
+        "db_api_port": {
```
Should we add a default here as well? We usually only use 27017.

GernotMaier replied:
Done.

orelgueta reviewed (commented):
Skimmed through it, added one small comment, not blocking.

GernotMaier commented:
@EshitaJoshi  - could you have another look? I've fixed the schema url and added the suggestion by Orel. Note that I have to look into the other schema in src/simtools/schema for up-to-date schema urls. I do that in a separate PR.

EshitaJoshi reviewed (approved):
Looks good, thanks for the PR!
